uint8 bid
uint8 mid
uint8 len
uint8 check  